velox/functions/lib/Re2Functions.cpp

```diff
@@ -938,8 +938,15 @@ std::shared_ptr<exec::VectorFunction> makeLike(
 
     auto constantEscape = escape->as<ConstantVector<StringView>>();
 
-    // Escape char should be a single char value
-    VELOX_USER_CHECK_EQ(constantEscape->valueAt(0).size(), 1);
+    try {
```

Comment:
Should the above check for escape != nullptr also use AlwaysFailingVectorFunction ?

Reply:
This is an interesting question. I think no. LIKE requires that 'escape' argument is a constant string and that check verifies it is "constant". It feels that this is part of signature (similar to types and number of arguments) rather than input data.